source/trigger/cdc.go

```diff
@@ -0,0 +1,353 @@
+	}
+
+	operationType, ok := row[colOperation].(string)
+	if !ok {
+		return sdk.Record{}, errWrongTrackingOperationType
+	}
+
+	// set a new position into the variable,
+	// to avoid saving position into the struct until we marshal the position
+	position := &Position{
+		Mode: ModeCDC,
+		// set the value from columnTrackingID column of the tracking table
+		LastProcessedVal: row[colID],
+		CreatedAt:        iter.position.CreatedAt,
+	}
+
+	convertedPosition, err := position.marshal()
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("convert position %w", err)
+	}
+
+	key := make(sdk.StructuredData)
+	if iter.key != "" {
+		val, ok := row[iter.key]
+		if !ok {
+			return sdk.Record{}, fmt.Errorf("key column %q not found", iter.key)
+		}
+
+		key[iter.key] = val
+	}
+
+	// delete tracking columns
+	delete(row, colOperation)
+	delete(row, colID)
+
+	rowBytes, err := json.Marshal(row)
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("marshal row: %w", err)
+	}
+
+	iter.position = position
+
+	metadata := sdk.Metadata{
+		common.MetadataPostgresTable: iter.table,
+	}
+	metadata.SetCreatedAt(time.Now())
+
+	switch operationType {
+	case actionInsert:
+		return sdk.Util.Source.NewRecordCreate(
+			convertedPosition,
+			metadata,
+			key,
+			sdk.RawData(rowBytes),
+		), nil
+	case actionUpdate:
+		return sdk.Util.Source.NewRecordUpdate(
+			convertedPosition,
+			metadata,
+			key,
+			nil,
+			sdk.RawData(rowBytes),
+		), nil
+	case actionDelete:
+		return sdk.Util.Source.NewRecordDelete(
+			convertedPosition,
+			metadata,
+			key,
+		), nil
+	default:
+		return sdk.Record{}, errWrongTrackingOperationType
+	}
+}
+
+// Close closes database rows of CDC iterator.
+func (iter *CDC) Close() (err error) {
+	// send a signal to stop clearing the tracking table
+	iter.tableSrv.stopCh <- struct{}{}
+
+	if iter.rows != nil {
+		iter.rows.Close()
+	}
+
+	select {
+	// wait until clearing tracking table will be finished
+	case <-iter.tableSrv.canCloseCh:
+	// or until time out
+	case <-time.After(timeoutBeforeCloseDBSec * time.Second):
+	}
+
+	iter.tableSrv.close()
+
+	return
+}
+
+// loadRows selects a batch of rows from a database, based on the
+// table, columns, orderingColumn, batchSize and the current position.
+func (iter *CDC) loadRows(ctx context.Context) error {
+	sb := psql.Select(iter.columns...).
+		From(iter.conduit).
+		OrderBy(colID).
+		Limit(iter.batchSize)
+
+	if iter.position.LastProcessedVal != nil {
+		sb = sb.Where(sq.Gt{colID: iter.position.LastProcessedVal})
+	}
+
+	query, args, err := sb.ToSql()
+	if err != nil {
+		return fmt.Errorf("create select query from tracking table: %w", err)
+	}
+
+	iter.rows, err = iter.conn.Query(ctx, query, args...)
+	if err != nil {
+		return fmt.Errorf("execute select query from tracking table: %w", err)
+	}
+
+	return nil
+}
+
+// newTrackingTableService is a service to delete processed rows.
+func newTrackingTableService() *trackingTableService {
+	stopCh := make(chan struct{}, 1)
```

Comment:
you can allocate variables inside trackingTableService initialization
```
return &trackingTableService{
		stopCh:     make(chan struct{}, 1),
		canCloseCh:  make(chan struct{}, 1),
		errCh:       make(chan error, 1),
		idsToDelete: make([]any, 0),
	}
```

Reply:
Perfect 🥇 
Done.